src/d3a/models/area/__init__.py

```diff
@@ -59,6 +59,39 @@
 )
 
 
+def check_area_name_exists_in_parent_area(parent_area, name):
+    """
+    Check the children of parent area , iterate through its children and
+        check if the name to be appended does not exist
+    Note: this check is to be called before adding a new area of changing its name
+    :param parent_area: Parent Area
+    :param name: New name of area
+    :return: boolean
+    """
+    for child in parent_area.children:
+        if child.name == name:
+            return True
+    return False
+
+
+class AreaChildrenList(list):
```

Comment:
I am always intimidated when extending the default Python types, because of any potential side effects that these might have. I think a much better approach would be a validator on the constructor of the class, which already has the `children` member populated. IMO this will be much simpler and less error-prone than a new datatype.

Reply:
@spyrostz The problem I am trying to address is the `append` or `insert` or changing a list's parent ( We can eliminate this option for now ) . So for the formers I don't see any other fix

Reply:
Have a look at the Area class constructor. The children constructor argument is a list of children for this area. Therefore, you can easily scan the list names for duplicates directly in the constructor of the area class, and raise the exception there. There is no need to override the insert / append of the list itself.

Reply:
@spyrostz This is only when creating the Area object , we initialize its children , and when we later do `area.children.append(child)` no constructor call is being made